packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -1469,7 +1470,7 @@ export function renderDidSuspendDelayIfPossible(): void {
 }
 
 export function renderDidError() {
-  if (workInProgressRootExitStatus !== RootCompleted) {
+  if (workInProgressRootExitStatus !== RootSuspendedWithDelay) {
```

Comment:
I was thinking if this state machine gets any more complicated, we could sort this enum in order of precedence. I suspect the precedence is always order independent so it feels like that should work.

Reply:
Why was it checking if completed in the first place? 😮

Reply:
Idk, defensive coding? :D